contractions_BSM.c

```diff
@@ -141,1746 +169,1889 @@ extern int nstore;
 
                }
+               else{
+                 for(src_idx = 0; src_idx < 12; src_idx++ )
+                 {
+                    snprintf(prop_fname,200,"bsm2prop.%.4d.%.2d.%02d.%.8d.inverted",nstore, isample, src_idx, iscalar);
 
-            }
-/*   
-       TYPE VI.1.a OR  VI.1.b   S(ytilde, x)    *phi^dagger(x-0)*tau_i*                    (1-gamma5)/2*S(x,ytilde)
-       TYPE VI.2.a OR  VI.2.b   S(ytilde, x-2*0)*phi^dagger(x-0)*tau_i*  U0(x-2*0)*U0(x-0)*(1-gamma5)/2*S(x,ytilde)
-*/
+                    for(pos = 0; pos < 8; ){
+                       printf("READCHECK: Propagator in pos %02d from file %s\n", pos/2,prop_fname);
 
-/**************************
-Multiplication with Stilde(x-2*0,ytilde)P(x) in three steps:
-1; Gathering P(x) from direction +0
-2; Multiplying Stilde(x-O,ytilde) with P(x+0)
-3; Gathering The result in direction -0
+//read the propagator from source d to sink d 
+                       read_spinor(g_spinor_field[0], g_spinor_field[1], prop_fname, pos);
+                       convert_eo_to_lexic(temp_field[0], g_spinor_field[0], g_spinor_field[1]);
+                       pos+=1;
 
-**************************/
+//read the propagator from source d to sink u
+                       read_spinor(g_spinor_field[0], g_spinor_field[1], prop_fname, pos);
+                       convert_eo_to_lexic(temp_field[1], g_spinor_field[0], g_spinor_field[1]);
+                       pos+=1;
 
+//create a bispinor first insert sink u then sink d
+//Store them in such a way that the u-ones should come first
+                       compact(operator_list[op_id].prop_zero[pos > 4 ? src_idx*4+pos/2-3 : src_idx*4+pos/2+ 1], temp_field[1], temp_field[0]);
+                    }
 
-            if ( type_12 == TYPE_2 ) {
-              for (c1=0; c1<3; ++c1){
-                 count=0;
-                 generic_exchange_direction_nonblocking( running2d[12*f1+3*s1+c1], sizeof(bispinor), TUP, request, &count );
-                 MPI_Waitall( count, request, statuses);
-              }
-            }
-            for (ix = 0; ix< VOLUME; ++ix){
-               for (c1=0; c1<3; ++c1){
-                  if (type_12 == TYPE_2){
-                    multiply_backward_propagator(&tmpbisp2d[12*f1+3*s1+c1][ix], propfields, &running2d[12*f1+3*s1+c1][g_iup[ix][TUP]], ix, TDOWN);
-                  }
-                  else if (type_12 == TYPE_1){
-                    multiply_backward_propagator(&running2d[12*f1+3*s1+c1][ix], propfields, &running2d[12*f1+3*s1+c1][ix]            , ix, NODIR);
-                  }
+                 }//end of loop for spinor and color source degrees of freedom
+
+                 if ( (vectorcurrentcurrent_BSM == 1 ) || ( axialcurrentcurrent_BSM == 1 )){
+                   for(src_idx = 0; src_idx < 12; src_idx++ )
+                   {
+                      snprintf(prop_fname,200,"bsm2prop.%.4d.%.2d.%02d.%.8d.inverted",nstore, T_global-1, src_idx, iscalar);
```

Comment:
@pittlerf What does `T_globa-1` encode here in the `isample` position?

Reply:
Hi, yeah, sorry yes, I just thought tat in the case of point sources that we use (just one at (0,0,0)), this might be the place, but  of coarse, it should have a standalone place.